src/services/item/plugins/published/plugins/search/index.ts

```diff
@@ -13,19 +16,27 @@ export type SearchFields = {
 };
 
 const plugin: FastifyPluginAsync = async (fastify) => {
-  const searchService = new SearchService();
-
-  fastify.decorate('search', { service: searchService });
+  const searchService = fastify.search.service;
 
-  // search for items with keyword
-  // range: title, tag, all, author
-  fastify.get<{ Querystring: SearchFields }>(
+  fastify.post(
     '/collections/search',
-    { schema: search, preHandler: fastify.attemptVerifyAuthentication },
-    async ({ member, query, log }) => {
-      return searchService.search(member, buildRepositories(), query);
+    { preHandler: fastify.attemptVerifyAuthentication, schema: search },
+    async ({ member, body }) => {
+      return searchService.search(member, buildRepositories(), body as MultiSearchParams);
     },
   );
+
+  fastify.get('/collections/search/rebuild', async ({ headers }, reply) => {
+    // TODO: in the future, lock this behind admin permission and maybe add a button to the frontend admin panel
+    const headerRebuildSecret = headers['meilisearch-rebuild'];
+
+    if (MEILISEARCH_REBUILD_SECRET && MEILISEARCH_REBUILD_SECRET === headerRebuildSecret) {
```

Comment:
It might be, but we don't have any secret management tool, so how would we ( or future devs) know the unhashed secret to send? This way it's simpler for now and we can find the secret on AWS (like the DB password)